### ISSUE TITLE
NuGetAuthenticate@0 is unable to install credential provider on force reinstall.

### ISSUE THREAD

**Issue by WardenGnaw** [microsoft/azure-pipelines-tasks#13178]
*Type*: Bug

**Enter Task Name**: NuGetAuthenticate

https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/NuGetAuthenticateV0

## Environment
- Server - Azure Pipelines or TFS on-premises? Azure Pipelines 
   Account Name: DevDiv
   Team Project Name: Concord
   Build Definition Name (Build Number): Concord-Windows-PR (11789)

- Agent - Private Windows (2.165.0)

## Issue Description
The error seen is:
```
[error]Error: Failed to remove the directory 'C:\Users\dlab14\.nuget\plugins\netfx\CredentialProvider.Microsoft' while installing the credential provider. Ensure that this directory is not in use and that the agent account has permission to delete this directory.
Error: EPERM: operation not permitted, unlink 'C:\Users\dlab14\.nuget\plugins\netfx\CredentialProvider.Microsoft\PowerArgs.dll'
```
This is due to:
Basically the bug is that if `forceReinstallCredentialProvider`  == true, the task should kill their own running exe before trying to overwrite DLLs it’s loaded and it itself.  That or maybe it was supposed to exit on its own, one of those is the bug.

**Comment by jmyersmsft** [comment]
The credential provider should exit on its own either when commanded to by NuGet, or if not, when the NuGet process exits. Is it possible that the machines in your pool are running multiple instances of the agent?

**Comment by WardenGnaw** [comment]
"Not on any of the machines I looked at while investigating. They should acknowledge there is always a chance a process hangs?"

**Comment by mattgrande** [comment]
@WardenGnaw - I'm getting this intermittently on one of my repos. Did you ever figure out why it was happening?

**Comment by WardenGnaw** [comment]
For us, we worked around the issue by rebuilding/redeploying machine and it has not gotten into that state for a while. 

However, we think the issue should be there because it does not stop `nuget` before overwriting its own DLLs.

**Comment by mattgrande** [comment]
I ended up bumping the timeout from (the default) 5s to 60s and haven't had another failure since, so hopefully that's all it needed?

**Comment by jmyersmsft** [comment]
We can't kill nuget or the cred provider from the task, because there may be multiple agents running builds on the same machine, and we have no way of knowing which processes correspond to this agent. If we did, we'd risk killing other unrelated builds.

**Comment by dirtygooback** [comment]
> For us, we worked around the issue by rebuilding/redeploying machine and it has not gotten into that state for a while.
> 
> However, we think the issue should be there because it does not stop `nuget` before overwriting its own DLLs.

And how does one go about changing the default timeout?
